<a href="https://colab.research.google.com/github/mahmoudabdelaziz120/FLYRANK_ML-1/blob/main/w01_research_question.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# ML-02 — Research Question and Provisional Lane

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

Lane 2 — Refresh / Content Opportunity Scoring.

What interested me most is the prioritization problem. When a team has thousands of pages and limited time, reviewing everything is not realistic.

## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

**Research question:**
Which pages should the content team review first.

**Unit of analysis:**
A single content page.

**Output:**
A ranked priority list of pages, with higher-priority pages reviewed first.

**Decision:**
Which pages deserve the team's attention first?

**Action:**
The content team can review the highest-ranked pages and decide whether to refresh, expand, protect, prune, or monitor them.

**Cost of a wrong recommendation:**
A wrong recommendation may waste limited team time on a low-value page while a more important opportunity is ignored.

**Why data or ML can help:**
The decision depends on multiple signals that may interact in complex ways. Data and ML may help identify patterns that are difficult to capture using fixed hand-written rules alone.


## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [11]:
import os

print(os.getcwd())
print(os.path.exists("data/raw/content_refresh_anonymized.csv"))

/content/FLYRANK_ML-1
True


In [12]:
declining = df["clicks_last_30d"] < df["clicks_prev_30d"]

print("Number of pages:", len(df))
print("Number of clients:", df["client_id"].nunique())
print("Pages with declining clicks:", declining.sum())
print("Declining pages %:", round(declining.mean() * 100, 2))

Number of pages: 30000
Number of clients: 32
Pages with declining clicks: 6806
Declining pages %: 22.69


The starter dataset contains **30,000 pages across 32 clients**.

I found that **6,806 pages (22.69%)** had lower clicks in the last 30 days compared with the previous 30 days.

This means that nearly **1 in 4 pages** showed an observed decline in clicks during this comparison period. With thousands of pages to review, prioritizing which pages deserve attention first becomes a practical decision problem.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

## 4. Careful words: what I can and can't claim

I can use the starter data to identify observed patterns and signals associated with changes in page performance and explore whether they can support page prioritization.

I cannot claim that these signals directly cause ranking or traffic decline, or that the analysis predicts Google's algorithm.

## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.